app/Extensions/Gutenberg/Blocks/Product.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace BlazeWooless\Extensions\Gutenberg\Blocks;
+
+use BlazeWooless\TypesenseClient;
+
+class Product {
+	private static $instance = null;
+
+	public static function get_instance() {
+		if ( self::$instance === null ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function __construct() {
+		add_action( 'save_post_gp_elements', array( $this, 'upsert_site_product' ) );
+		add_action( 'blaze_wooless_after_site_info_sync', array( $this, 'sync_product_to_typesense' ) );
+	}
+
+	public function sync_product_to_typesense() {
+
+		$post_id = $this->get_site_product_id();
+		if ( ! empty( $post_id ) ) {
+
+			$site_product = $this->get_site_product( $post_id );
+
+			do_action(
+				"inspect",
+				[ 
+					"sync_product_to_typesense",
+					[ 
+						"post_id" => $post_id,
+						"site_product_id" => $site_product
+					]
+				]
```

Comment:
avoid using the shortcut version of the array. use array() instead of []. See https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays